third-party/cmake/BuildLuv.cmake

```diff
@@ -62,13 +62,18 @@ set(LUV_INCLUDE_FLAGS
 set(LUV_PATCH_COMMAND
     ${CMAKE_COMMAND} -E copy_directory ${LUV_SRC_DIR}/rockspecs ${LUV_SRC_DIR})
 
+if(NOT LUV_LUA_ENGINE)
```

Comment:
check `PREFER_LUA` here instead?

Reply:
I thought about this, too, but I wasn't sure if that should also influence stuff like `USE_BUNDLED_LUAJIT` and `USE_BUNDLED_LUA`. What do you think?

Also, just as a side note, `PREFER_LUA` is currently used only in the main CMakeLists, not the third-party one. So e.g. in `local.mk` you would still have to specify it twice, for `CMAKE_EXTRA_FLAGS` and for `DEPS_CMAKE_FLAGS`.

Reply:
having a single variable (now that it exists anyway) seems like a good idea. I can't imagine a case where you would want luajit for luv and lua for the rest. 
Luajit tends to be the favorite, except when it's not available basically.

Reply:
Yes, `PREFER_LUA` is a bit unspecific / undefined currently (and only for the main project).

I think it might make sense to factor out the handling of it into a shared cmake file, and then also use it for third-party.  Related code: https://github.com/neovim/neovim/blob/38342d75f64e0825505bbb6ff6e1a67b12bf3f17/CMakeLists.txt#L392-L402
(not necessarily controlling it through `PREFER_LUA`, but could work similar as with BuildLuarocks)

As for the issue at hand: `-DWITH_LUA_ENGINE` should be set to `Lua` maybe with `-`USE_BUNDLED_LUAJIT=0` / `USE_BUNDLED_LUA=1` (is this what you are using, or do you want to use the system Lua?)

```diff
diff --git i/third-party/cmake/BuildLuv.cmake w/third-party/cmake/BuildLuv.cmake
index 9f8e4bbc4..967e0a171 100644
--- i/third-party/cmake/BuildLuv.cmake
+++ w/third-party/cmake/BuildLuv.cmake
@@ -71,6 +71,19 @@ set(LUV_CONFIGURE_COMMAND_COMMON
   -DBUILD_SHARED_LIBS=OFF
   -DBUILD_MODULE=OFF)

+if(USE_BUNDLED_LUAJIT)
+  list(APPEND LUV_CONFIGURE_COMMAND_COMMON -DWITH_LUA_ENGINE=LuaJit)
+elseif(USE_BUNDLED_LUA)
+  list(APPEND LUV_CONFIGURE_COMMAND_COMMON -DWITH_LUA_ENGINE=Lua)
+else()
+  find_package(LuaJit)
+  if(LUAJIT_FOUND)
+    list(APPEND LUV_CONFIGURE_COMMAND_COMMON -DWITH_LUA_ENGINE=LuaJit)
+  else()
+    list(APPEND LUV_CONFIGURE_COMMAND_COMMON -DWITH_LUA_ENGINE=Lua)
+  endif()
+endif()
+
 if(USE_BUNDLED_LIBUV)
   set(LUV_CONFIGURE_COMMAND_COMMON
     ${LUV_CONFIGURE_COMMAND_COMMON}
```

Reply:
=> https://github.com/neovim/neovim/pull/10449